### DISCUSSION TITLE
forrtl: error (65): floating invalid. MODFLOW-6 simulation error related to DISV package

### DISCUSSION THREAD

**Discussion by jarmodevries** [modflowpy/flopy#2394] [Q&A] [answered]
### Dear community

I'm trying to run a MODFLOW model in FloPy using an unstructured grid I constructed in ArcGIS Pro. 
I've imported this grid as a shapefile. Extracting polygon coordinates from the shapefile, I wrote a DISV package, without any errors. Here's a plot showing part of the grid:

![afbeelding](https://github.com/user-attachments/assets/0107d10f-a4f1-4d83-88c0-800804416f55)

Writing the other packages also went well: it all gives input files which seem to look okay. 

When running the entire simulation, I get this error:
```
MODFLOW runs in SEQUENTIAL mode
 
 Run start date and time (yyyy/mm/dd hh:mm:ss): 2024/12/19 14:13:51
 

 Writing simulation list file: mfsim.lst
 Using Simulation name file: mfsim.nam
 
forrtl: error (65): floating invalid
Image              PC                Routine            Line        Source             
mf6.exe            00007FF6432EF20A  DISVGEOM_mp_CPROP         193  DisvGeom.f90
mf6.exe            00007FF643296135  CONNECTIONSMODULE         804  Connections.f90
mf6.exe            00007FF6432D3CDE  DISVMODULE_mp_CON         696  Disv.f90
mf6.exe            00007FF6432D249B  DISVMODULE_mp_GRI         511  Disv.f90
mf6.exe            00007FF64349195F  GWFMODULE_mp_GWF_         228  gwf.f90
mf6.exe            00007FF6435A1E4C  MF6COREMODULE_mp_         318  mf6core.f90
mf6.exe            00007FF64359E0CB  MF6COREMODULE_mp_          43  mf6core.f90
mf6.exe            00007FF64359DDB2  MAIN__                     11  mf6.f90
mf6.exe            00007FF6437D956E  Unknown               Unknown  Unknown
mf6.exe            00007FF6437D9950  Unknown               Unknown  Unknown
KERNEL32.DLL       00007FFBF0747374  Unknown               Unknown  Unknown
ntdll.dll          00007FFBF211CC91  Unknown               Unknown  Unknown
Model did not run successfully.
```

Does anyone know where this error comes from and if there are any ways to fix it? I suspect the error has something to do with the DISV package, but I checked the coordinates (possible duplicates?), cell areas, number of vertices, etc, and that all looks okay...

**Comment by christianlangevin** [comment, marked as answer]
What version of MODFLOW 6 are you using?  MODFLOW is unforgiving with the vertices and cell definitions.  Have you tried your workflow with a very simple grid, perhaps consisting of just a couple cells?  If you still can't find the issue, feel free to upload a minimal example that demonstrates the problem and we can take a look.

**Comment by jarmodevries** [comment]
Dear Christian,

Thanks for your response! I followed your approach by running small parts of the grid one by one. The error turned out to be a hole in the grid, which I hadn't noticed initially because the grid is so large. Filling it up with an extra cell solved the issue.


![Screenshot 2025-01-14 105259](https://github.com/user-attachments/assets/4929878a-283f-45a8-9d36-97522a80bb1f)
